man/crypttab.xml

```diff
@@ -431,6 +431,25 @@
         </para></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><option>x-initrd.attach</option></term>
+
+        <listitem><para>Setup this encrypted block device in the initramfs, similarly to
+        <citerefentry><refentrytitle>systemd.mount</refentrytitle><manvolnum>5</manvolnum></citerefentry>
+        units marked with <option>x-initrd.mount</option>.</para>
+
+        <para>Although it's not necessary to mark the mount entry for the root file system with
```

Comment:
I can't say I like the general recommendation of this paragraph. I agree with it is OK to add this mechanism so it can be used in the corner cases that we can't figure out, but recommending to people to use this pretty much always seems like a step in the wrong direction. Maybe weaken this by saying "This option may be added to any block device that contains a file system which should not be unmounted before switching back to the initramfs during shutdown."

Reply:
Well unless we start trying to inherit x-initrd.mount in fstab to crypttab somehow we won't be able to figure out which encrypted devices need the option I think. And I don't think we want to do that, do we ?

So it might better at the end to recommend marking all relevant devices rather than documenting that some of them get the option automatically but not the others.

Reply:
> unless we start trying to inherit x-initrd.mount in fstab to crypttab somehow we won't be able to figure out which encrypted devices need the option I think. And I don't think we want to do that, do we ?

I'd say that is exactly what we want to do. It might not be easy to implement, but if possible, why not?

Reply:
That reminds me a old but similar discussion where inheriting options from crypttab to fstab wasn't a goo idea: https://github.com/systemd/systemd/pull/2979

Reply:
#2979 was in the opposite direction: propagating settings from low-level units to higher level ones. The upstart model, if you so will. The still support the conclusion to turn that down.

Reply:
right.